addons/project/models/project.py

```diff
@@ -128,9 +128,22 @@ def unlink_wizard(self, stage_view=False):
             'context': context,
         }
 
+    @api.model
+    def create(self, vals):
+        if 'user_id' in vals:
+            vals.pop('project_ids', None)
+        return super(ProjectTaskType, self).create(vals)
+
     def write(self, vals):
         if 'active' in vals and not vals['active']:
             self.env['project.task'].search([('stage_id', 'in', self.ids)]).write({'active': False})
+        if 'user_id' in vals and vals['user_id']:
+            vals.pop('project_ids', None)
+            if self.project_ids:
+                self.project_ids = None
+        if 'project_ids' in vals and self.user_id:
+            self.user_id = None
```

Comment:
Same comment here. 

Moreover, I think the `api.constrains` should be easier.

Something like: 

```py
@api.constrains('user_id', 'project_ids')
def _check_personal_stage_not_linked_to_projects(self):
    if any(stage.user_id and stage.project_ids for stage in self):
        raise UserError('A personal stage cannot be linked to a project because that stage is only visible to the user set.')
```

Also can you define that constrains **below** the fields definition.

I will validate the user error message by @aju-odoo . 🙂

Reply:
Ok thanks for the tips! Much easier indeed :)
I'll wait for the confirmation of the message and add it to the .pot file :)

Reply:
Don't know if the tests are really necessary when using the constraint, but I kept them and modified them :) Let me know if you don't think it is necessary :)

Reply:
Yes, I prefer we keep the test to be sure the constrains is always applied. 🙂 

The error message will be the following one: `A personal stage cannot be linked to a project because it is only visible to its corresponding user.` 🙂